### PR TITLE
fix setHandleSurfaceEvent

### LINKED ISSUE #7: Can we use minSdkVersion = 9?
I have a project that needs to support old devices, can this lib be used on sdk = 9

### PR DESCRIPTION
**Pull request by sibelius**
This commit update gradle plugins and lower the minSdkVersion
requirements to 9
closes ragnraok/RxCamera#7

### CODE REVIEW

**Comment by sibelius** [comment]
I've tried to take if off .idea folder from this repo using .gitignore but it didn't work it out

**Comment by sibelius** [comment]
fix setHandleSurfaceEvent
fix ragnraok/RxCamera#5, fix ragnraok/RxCamera#10
set minSdkVersion 14

I decided to drop support to older version, so we can have a FaceDetection in this lib (https://github.com/ragnraok/RxCamera/issues/9)

**Comment by ragnraok** [comment]
It looks weird, now you have two `onAvailable` with different parameter, and you just want to set the preview surface, but I already set it when you bind the surface

File: lib/src/main/java/com/ragnarok/rxcamera/RxCameraInternal.java [hunk 1]
```diff
@@ -298,6 +300,7 @@ public boolean bindSurfaceInternal(SurfaceView surfaceView) {
         try {
             bindSurfaceView = surfaceView;
             if (cameraConfig.isHandleSurfaceEvent) {
+                surfaceCallback.setSurfaceListener(this);
```
Comment (sibelius) [inline lib/src/main/java/com/ragnarok/rxcamera/RxCameraInternal.java:303]: @ragnraok you didn't add a listener to the surfaceCallback

File: lib/src/main/java/com/ragnarok/rxcamera/SurfaceCallback.java [hunk 2]
```diff
@@ -10,13 +10,14 @@
 /* package */ class SurfaceCallback implements SurfaceHolder.Callback, TextureView.SurfaceTextureListener {
 
     public interface SurfaceListener {
-        void onAvailable();
+        void onAvailable(SurfaceHolder holder);
+        void onAvailable(SurfaceTexture surface);
```
Comment (sibelius) [inline lib/src/main/java/com/ragnarok/rxcamera/SurfaceCallback.java:14]: We could use a template as well here, just to have one method

File: lib/src/main/java/com/ragnarok/rxcamera/RxCameraInternal.java [hunk 3]
```diff
@@ -381,9 +385,10 @@ public boolean closeCameraInternal() {
     }
 
     @Override
-    public void onAvailable() {
+    public void onAvailable(SurfaceHolder holder) {
         if (isNeedStartPreviewLater) {
             try {
+                camera.setPreviewDisplay(holder);
```
Comment (sibelius) [inline lib/src/main/java/com/ragnarok/rxcamera/RxCameraInternal.java:391]: you forgot to add the previewDisplay when the Surface was created, or the Texture was available, this is important if you don't have an open button, and you want to preview as soon as the surface/texture is available

**Comment by ragnraok** [comment]
Ok.... I know that, It seem that I forgot to call `setSurfaceListener` in `RxCameraInternal`, so the `setHandleSurfaceEvent` doesn't work, I will fix it latter, thanks for your pull request

**Comment by sibelius** [comment]
this PR fix this, I am using it on a production app.
Thanks for this lib, really make sense to have a Rx API for this.

I will implement this issue https://github.com/ragnraok/RxCamera/issues/9 as well, are you willing to add another PR for this repo?

**Comment by ragnraok** [comment]
It's OK, you can used the `Camera.FaceDetectionListener` or implement the face detection with your algorithm

**Pull request merged**